src/main/java/org/cactoos/func/AsyncFunc.java

```diff
@@ -91,12 +92,12 @@ public AsyncFunc(final Proc<X> proc, final ThreadFactory fct) {
      */
     public AsyncFunc(final Func<X, Y> fnc, final ThreadFactory fct) {
         this.func = fnc;
-        this.factory = fct;
+        this.executor = Executors.newSingleThreadExecutor(fct);
```

Comment:
@caarlos0 I believe that this ctor must become a secondary one. We should introduce a primary one, which will accept `Func` and `ExecutorService`.